Automattic/wp-calypso
# Gridicons: Rename gridiconFormatChecker to check-gridicon-format

## Background
The file that performs pre-commit checks for standard Gridicon sizes in `bin` directory had an inconsistent naming convention of `gridiconFormatChecker`. This PR is to update the file name to an appropriate convention as `check-gridicon-format`.

Additionally, also update the usage in the `bin\pre-commit` hook to ensure the Gridicon size checks are correctly performed during file commits.

## Testing
1. Make sure to install the pre-commit hooks as outlined in [JavaScript Coding Guidelines](https://wpcalypso.wordpress.com/devdocs/docs/coding-guidelines/javascript.md).
2. Try to commit a Gridicon with a nonstandard size (without indicating it as a nonstandard Gridicon) to ensure the pre-commit checks for Gridicons is still working.

 For e.g., Change the size attribute of some Gridicon tag in the following file `client/components/gridicon/docs/example.jsx` from `size={ 48 }` to `size={ 27 }`.

 Note: You can change it to any non-standard value. The list of standard values is `` as specified in ``.
3. Now stage the file for commit.
4. Now try to commit the file, this should run the pre-commit checks and produce the following commit failure output.

>eslint validation complete
[31mNon-standard gridicon size (  27 px ) detected in client/components/gridicon/docs/example.jsx line 24
>
[mValid gridiconsizes are 12px, 18px, 24px, 36px, 48px, 54px, 72px
[mIf you need any of those Gridicons to have a non-standard size, please, add 'nonStandardSize' as a property. See https://github.com/Automattic/wp-calypso/tree/master/client/components/gridicon#props
>
[31mGridicon Format Check Failed: [0mclient/components/gridicon/docs/example.jsx
>
[41mCOMMIT FAILED:[0m Your commit contains files that should pass validation tests but do not. Please fix the errors and try again.

## Reviewers
@johnHackworth @stephanethomas @nb 

## Issues
Closes #1821
